package-system/Lua/FindLua.cmake.template

```diff
@@ -17,19 +17,22 @@ set(MY_NAME "Lua")
 set($${MY_NAME}_INCLUDE_DIR $${CMAKE_CURRENT_LIST_DIR}/Lua/include)
 set($${MY_NAME}_LIBS_DIR $${CMAKE_CURRENT_LIST_DIR}/Lua/lib)
 
-set($${MY_NAME}_LIBRARY_DEBUG   $${$${MY_NAME}_LIBS_DIR}/debug/$${CMAKE_STATIC_LIBRARY_PREFIX}lualib$${CMAKE_STATIC_LIBRARY_SUFFIX})
 set($${MY_NAME}_LIBRARY_RELEASE $${$${MY_NAME}_LIBS_DIR}/release/$${CMAKE_STATIC_LIBRARY_PREFIX}lualib$${CMAKE_STATIC_LIBRARY_SUFFIX})
+set($${MY_NAME}_LIBRARY_DEBUG   $${$${MY_NAME}_LIBS_DIR}/debug/$${CMAKE_STATIC_LIBRARY_PREFIX}lualib$${CMAKE_STATIC_LIBRARY_SUFFIX})
 
 # we set it to a generator expression for multi-config situations:
 set($${MY_NAME}_LIBRARY  
     "$$<$$<CONFIG:profile>:$${$${MY_NAME}_LIBRARY_RELEASE}>"
     "$$<$$<CONFIG:Release>:$${$${MY_NAME}_LIBRARY_RELEASE}>"
     "$$<$$<CONFIG:Debug>:$${$${MY_NAME}_LIBRARY_DEBUG}>")
 
-add_library($${TARGET_WITH_NAMESPACE} INTERFACE IMPORTED GLOBAL)
+add_library($${TARGET_WITH_NAMESPACE} STATIC IMPORTED GLOBAL)
+set_target_properties($${TARGET_WITH_NAMESPACE} PROPERTIES IMPORTED_LOCATION "$${$${MY_NAME}_LIBRARY_RELEASE}")
+set_target_properties($${TARGET_WITH_NAMESPACE} PROPERTIES IMPORTED_LOCATION_DEBUG "$${$${MY_NAME}_LIBRARY_DEBUG}")
 ly_target_include_system_directories(TARGET $${TARGET_WITH_NAMESPACE} INTERFACE $${$${MY_NAME}_INCLUDE_DIR})
+
 target_link_libraries($${TARGET_WITH_NAMESPACE} 
-    INTERFACE $${$${MY_NAME}_LIBRARY}
+    INTERFACE $${$${MY_NAME}_LIBRARY} $${CMAKE_DL_LIBS}
```

Comment:
FYI the `$${$${MY_NAME}_LIBRARY}` isn't needed since the `IMPORTED_LOCATION` variable has been set.
```suggestion
    INTERFACE $${CMAKE_DL_LIBS}
```